src/core_modules/capture-core/metaDataMemoryStoreBuilders/trackedEntityTypes/factory/TrackedEntityType/TeiRegistrationFactory.js

```diff
@@ -79,12 +84,66 @@ export class TeiRegistrationFactory {
             o.id = Section.MAIN_SECTION_ID;
             o.name = i18n.t('Profile');
         });
-
         featureTypeField && section.addElement(featureTypeField);
+
+        if (this.dataEntryFormConfig && cachedTrackedEntityTypeAttributes) {
+            const trackedEntityAttributeDictionary = cachedTrackedEntityTypeAttributes
+                .reduce((acc, trackedEntityAttribute) => {
+                    if (trackedEntityAttribute.trackedEntityAttributeId) {
+                        acc[trackedEntityAttribute.trackedEntityAttributeId] = trackedEntityAttribute;
+                    }
+                    return acc;
+                }, {});
+
+            // $FlowFixMe
+            this.dataEntryFormConfig.asyncForEach(async (formConfigSection) => {
+                const attributes = formConfigSection.elements.reduce((acc, element) => {
+                    if (element.type === 'plugin') {
+                        const fieldMap = element
+                            .fieldMap
+                            ?.map(field => ({
+                                ...field,
+                                ...trackedEntityAttributeDictionary[field.IdFromApp],
+                            }));
+
+                        acc.push({ ...element, fieldMap });
+                        return acc;
+                    }
+                    const attribute = trackedEntityAttributeDictionary[element.id];
+                    if (attribute) {
+                        acc.push(attribute);
+                    }
+                    return acc;
+                }, []);
+
+                await attributes.asyncForEach(async (trackedEntityAttribute) => {
+                    if (trackedEntityAttribute?.id === 'plugin') {
+                        const element = new DataEntryPluginConfig((o) => {
+                            o.id = trackedEntityAttribute.id;
+                            o.name = trackedEntityAttribute.name;
+                            o.pluginSource = trackedEntityAttribute.pluginSource;
+                            o.fields = new Map();
+                        });
+
+                        await trackedEntityAttribute.fieldMap.asyncForEach(async (field) => {
+                            const dataElement = await this.dataElementFactory.build(field);
+                            dataElement && element.addField(field.IdFromPlugin, dataElement);
+                        });
+
+                        element && section.addElement(element);
+                    } else {
+                        const element = await this.dataElementFactory.build(trackedEntityAttribute);
+                        element && section.addElement(element);
+                    }
```

Comment:
This is the same logic as in `EnrollmentFactory.js`. Can it be extracted to a commonplace and reused?